lib/src/widgets/request_list_widget.dart

```diff
@@ -9,19 +9,22 @@ class RequestListWidget extends StatelessWidget {
   const RequestListWidget({
     super.key,
     required this.requestList,
-    required this.currentUserId,
+    required this.currentUser,
     required this.onRefresh,
     required this.onRequestSelected,
     required this.onVoteChange,
     this.onLongPress,
   });
   final List<RequestItem> requestList;
-  final String currentUserId;
+  final Creator currentUser;
   final Future<void> Function() onRefresh;
   final void Function(RequestItem) onRequestSelected;
   final void Function(RequestItem) onVoteChange;
   final void Function(RequestItem)? onLongPress;
 
+ bool isOwner(Creator creator) => creator.userId  == currentUser.userId
```

Comment:
An admin is not an owner and therefore this naming is confusing, If you want to include the "isAdmin" condition you should name it canEdit rather than isOwner.